functions/src/sections/Section.controller.ts

```diff
@@ -0,0 +1,43 @@
+// TODO: replace package import path
+import { DetailedClassInformation } from '@isaaccormack/uvic-course-scraper/dist/src/types';
+import { Get, Controller, Route, Response, Path, Query } from 'tsoa';
+import { Term } from '../constants';
+import { Section } from './Section.model';
+import { SectionsService } from './Section.service';
+
+@Route('sections')
+export class SectionsController extends Controller {
+  @Response(404, 'Section Not Found')
+  @Get('{term}')
+  public async sections(
+    @Path() term: Term,
```

Comment:
this throws a nasty validation error when the term isn't found, is there anyway to add some middleware to catch this error and give a nicer error back to the client like simply, invalid term. This is just a nice to have, feel free to say no.

Example of response for term=202000:

<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8">
	<title>Error</title>
</head>

<body>
	<pre>ValidateError: <br> &nbsp; &nbsp;at new ValidateError (/Users/i/Develop/clockwork/functions/node_modules/@tsoa/runtime/src/routeGeneration/templateHelpers.ts:791:5)<br> &nbsp; &nbsp;at getValidatedArgs (/Users/i/Develop/clockwork/functions/build/routes.ts:254:19)<br> &nbsp; &nbsp;at /Users/i/Develop/clockwork/functions/build/routes.ts:143:33<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/route.js:112:3)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at /Users/i/Develop/clockwork/functions/node_modules/express/lib/router/index.js:281:22<br> &nbsp; &nbsp;at param (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/index.js:354:14)<br> &nbsp; &nbsp;at param (/Users/i/Develop/clockwork/functions/node_modules/express/lib/router/index.js:365:14)</pre>
</body>

</html>

Reply:
There is a way but I haven't gotten around to it yet fully. It doesn't seem it should be a blocker though for this PR. In production that error isn't displayed to the client.

Reply:
Ok